scipy/special/_logit.h

```diff
@@ -3,12 +3,38 @@
 
 #include "numpy/npy_math.h"
 
-npy_float logitf(npy_float x);
-npy_double logit(npy_double x);
-npy_longdouble logitl(npy_longdouble x);
 
-npy_float expitf(npy_float x);
-npy_double expit(npy_double x);
-npy_longdouble expitl(npy_longdouble x);
+template <typename T>
+inline T _logit(T x) {
+    x /= 1 - x;
+    return npy_log(x);
+};
+
+
+template<typename T>
+inline T _expit(T x) {
+    return 1 / (1 + npy_exp(-x));
+};
+
+
+#ifdef __cplusplus
+#define EXTERN_C_START extern "C" {
+#define EXTERN_C_END }
+#else
+#define EXTERN_C_START
+#define EXTERN_C_END
+#endif
+
+EXTERN_C_START
```

Comment:
Since there are templates in this file, `ifdef __cplusplus` doesn't make sense. If including from c is desirable, then I would move the templates into a cpp file.

Reply:
Of course, fixed and force-pushed.